pkg/artifactory/resource/webhook/resource_artifactory_webhook.go

```diff
@@ -346,8 +372,15 @@ func ResourceArtifactoryWebhook(webhookType string) *schema.Resource {
 		return domainCriteriaValidationLookup[webhookType](ctx, criteria[0].(map[string]interface{}))
 	}
 
+	// taken from example in https://www.terraform.io/plugin/sdkv2/resources/state-migration#terraform-v0-12-sdk-state-migrations
+	var resourceSchemaV1 = func() *schema.Resource {
```

Comment:
Actually I'm not sure this needs to be a func. I think a var that has the `*schema.Resource` will work just the same

Reply:
Var is assignable. Func isn't

Reply:
I wasn't referring to `var`. Instead of this:

```go
var resourceSchemaV1 = func() *schema.Resource {
	return &schema.Resource{
		Schema: domainSchemaLookup(1)[webhookType],
	}
}

// then use this later:

Type:    resourceSchemaV1().CoreConfigSchema().ImpliedType(),
```

I think we should just have:

```go
resourceSchemaV1 := &schema.Resource{
	Schema: domainSchemaLookup(1)[webhookType],
}

// and

Type:    resourceSchemaV1.CoreConfigSchema().ImpliedType(),
```